File: TsogosunProfileAdminWeb/src/app/model/Dtos/profile-rewards-admin/scratchCardPrizeDto.ts
import { ScratchCardPrize } from "../../profile-rewards-admin/scratchCardPrize";

export interface ScratchCardPrizeDto extends ScratchCardPrize  {
    priceDescription?: string | undefined;
    winningMessage?: string | undefined;
    regretMessage?: string | undefined;    
    prizesAvailable: number;
    priceDetails?: string | undefined;

}

File: TsogosunProfileAdmin/MSPatronRewardsAdmin/Model/Dtos/ScratchCardPrizeDto.cs

namespace MSPatronRewardsAdmin.Model.Dtos
{
    public class ScratchCardPrizeDto : ScratchCardPrize
    {
        public string PriceDescription { get; set; }
        public string WinningMessage { get; set; }
        public string RegretMessage { get; set; }
        public int PrizesAvailable { get; set; }
        public string PriceDetails
        {
            get
            {
                return $"{PriceDescription}({NumberOfPrizes}) - {WinningMessage}";
            }
        }

    }
}
